stdlib/src/builtin/string.mojo

```diff
@@ -51,6 +51,23 @@ fn _ctlz(val: SIMD) -> __type_of(val):
     return llvm_intrinsic["llvm.ctlz", __type_of(val)](val, False)
 
 
+@always_inline
+fn _str_compare(str1: String, str2: String) -> Int:
+    var min_len = len(str1) if len(str1) < len(str2) else len(str2)
```

Comment:
Joe recently added `min` to a new `utils._math`, so we could import that in this file:

```mojo
from utils._math import min
```

and then use it here:

```suggestion
    var min_len = min(len(str1), len(str2))
```